### PR TITLE
Python SDK: Support multiple pydantic versions

### LINKED ISSUE #6903: Python SDK (lakefs_sdk) depends on outdated pydantic 1.X and prevents migration from lakefs_client
Hi,

All "our" Python projects depend in Pydantic 2.0 (which has been live since June 2023).
We tried to migrate lakefs_client to lakefs_sdk and it seems that Pydantic 1 is required. 

Would it be possible (given that lakefs_sdk is "new"), to use Pydantic V2?
https://pypi.org/project/pydantic/

This is our output from poetry
```
Because lakefs-sdk (1.0.0) depends on pydantic (>=1.10.5,<2)
```

PS: If it helps, copied from Pydantic doc

```
If you're using Pydantic V1 you may want to look at the [pydantic V1.10 Documentation](https://docs.pydantic.dev/) or, [1.10.X-fixes git branch](https://github.com/pydantic/pydantic/tree/1.10.X-fixes). Pydantic V2 also ships with the latest version of Pydantic V1 built in so that you can incrementally upgrade your code base and projects: from pydantic import v1 as pydantic_v1.
```

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #6903

### Background

Support pydantic v1/2 on python SDK and HL SDK.
      
### New Feature

We use templates to untag the pydantic version together with a post processing script which changes all imports to try and use the pydantic.v1 compatible module (in pydantic v2) and fallback on pydantic v1 modules

### Testing Details

Tested on HL SDK integration tests with pydantic v2 installed

### Breaking Change?

No

### CODE REVIEW

File: clients/python/scripts/pydantic.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+while IFS= read -r line; do
+    echo "... $line ..."
+    modules=$(echo $line | awk -F 'from pydantic import ' '{print $2}')
+    echo $line
+    
+    newline="try:\n    from pydantic.v1 import $modules\nexcept ImportError:\n    from pydantic import $modules"
+    
+    echo "$newline"
+    
+    echo "Replacing import statement"
```
Comment (itaiad200) [inline clients/python/scripts/pydantic.sh:14]: I think all the `echo`s are great for debugging but kind of redundant, maybe just output the file

**Review by arielshaqed — APPROVED** [review]
Neat!  Much value in few lines

1. How was this tested?
1. How can we test this on DataBricks?

**Review by ozkatz — COMMENTED** [review]
THANK YOU!

**Review by ozkatz — APPROVED** [review]

**Comment by N-o-Z** [comment]
> Neat! Much value in few lines
> 
> 1. How was this tested?
> 2. How can we test this on DataBricks?

So, I thought about doing it in a separate PR but since you asked. I created a matrix for the wrapper unit tests workflow.
One will run with pydantic v2 while the other with pydantic v1 (exact version taken from the AWS DBR 15 dependency - BTW in Azure DBR 12.2 there's no pydantic dep.)
However this will take affect only after we release a new SDK client and modify the HL SDK dep.

**Pull request merged**